bedrock/utils/braze.py

```diff
@@ -0,0 +1,43 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+from django.conf import settings
+
+import requests
+
+
+class BrazeClient:
+    def __init__(self, api_url_base, api_key) -> None:
+        self.api_key = api_key
+        self.track_user_api_url = f"{api_url_base}/users/track"
+
+    def subscribe(self, email, newsletter_id="news", source_url=None):
+        if newsletter_id not in settings.BRAZE_API_NEWSLETTERS:
+            raise RuntimeError("Invalid Newsletter ID")
+
+        attributes = {
+            "_update_existing_only": False,
+            "email": email,
+            "email_subscribe": "subscribed",
+            "user_alias": {
+                "alias_name": email,
+                "alias_label": "email",
+            },
+            "subscription_groups": [
+                {
+                    "subscription_group_id": settings.BRAZE_API_NEWSLETTERS[newsletter_id],
+                    "subscription_state": "subscribed",
+                }
+            ],
+        }
+        if source_url:
+            attributes["source_url"] = source_url
```

Comment:
I'll need to create this user attribute in the Braze UI before we deploy, let's coordinate on that when we merge this code!

Reply:
There is still some question of where this should go. Should it be a user attribute, or an attribute of an event that should be created?

Reply:
User attribute would be simplest, but becomes problematic if a user was to ever sign up for multiple newsletters (as the `source_url` would be re-set upon every newsletter signup).

Maybe an event is the most flexible approach 🤔